native/src/main/java/org/ballerinax/asb/listener/EntityConfig.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinax.asb.listener;
+
+/**
+ * {@code EntityConfig} represents an ASB entity configurations.
+ */
+public interface EntityConfig {
+}
```

Comment:
any reason to define this empty interface?

Reply:
This is used as a union type for `TopicConfig` and `QueueConfig` records.

Reply:
It seems you are using it as a marker interface. +1